packages/maps/src/components/result/addons/components/ScriptLoader.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect } from 'react';
+import { LoadScript } from '@react-google-maps/api';
+import { connect } from '@appbaseio/reactivesearch/lib/utils';
+import types from '@appbaseio/reactivecore/lib/utils/types';
+import {
+	SET_GOOGLE_MAP_SCRIPT_LOADING,
+	SET_GOOGLE_MAP_SCRIPT_LOADED,
+	SET_GOOGLE_MAP_SCRIPT_ERROR,
+} from '@appbaseio/reactivecore/lib/constants';
+
+const LIBRARIES = ['places'];
+
+const ScriptLoader = (props) => {
+	const {
+		children,
+		libraries,
+		mapKey,
+		setMapScriptLoadError,
+		setMapScriptLoaded,
+		setMapScriptLoading,
+		mapScriptLoadStatus,
+	} = props;
+
+	useEffect(() => {
+		if (mapScriptLoadStatus.error) {
+			console.error('Error loading google api: ', mapScriptLoadStatus.error);
+		}
+	}, [mapScriptLoadStatus.error]);
+
+	if (
+		!window.google
+		&& window.GOOGLE_SCRIPT_LOCK_ACQUIRED === true
+		&& !mapScriptLoadStatus.loaded
+	) {
+		return 'Initializing...';
+	}
+
+	if (!window.google && typeof window.GOOGLE_SCRIPT_LOCK_ACQUIRED === 'undefined') {
```

Comment:
@mohdashraf010897 We discussed to use Redux actions for it. You're just using variables here, which is incorrect. It won't re-render the component once the script is loaded.

Reply:
@bietkul redux actions alone are not enough. Think about the first render of the components(say ReactiveGoogleMap and GeoDistanceDropdown are used in one app together), at the initial render they both would be having the same value of the redux state, only the rerender would supply them with the updated value, and till then they both would have requested the google script individually.